### DISCUSSION TITLE
Stop consuming more bytes as soon as the final terminal is encountered?

### DISCUSSION THREAD

**Discussion by ssquirrel** [alecthomas/participle#277] [Q&A] [answered]
I am trying to embed participle into a larger parser. However it seems that participle will consume all the bytes from the reader. Is there a way to change this behavior? One way I can think of is supplying a custom lexer which emits fake EOF when needed.

**Comment by alecthomas** [comment, marked as answer]
Yep it's totally up to the Lexer. The builtin lexers do not support this at all, and the parser reads the entire token stream into memory before parsing.

**Reply by ssquirrel** [reply]
Thanks

**Reply by ssquirrel** [reply]
Also thank you so much for the amazing work!

**Reply by alecthomas** [reply]
You're welcome 🙂
